craft_providers/multipass/multipass.py

```diff
@@ -386,3 +387,65 @@ def umount(self, *, mount: str) -> None:
                 brief=f"Failed to unmount {mount!r}.",
                 details=errors.details_from_called_process_error(error),
             ) from error
+
+    def wait_until_ready(
+        self, *, retry_wait: float = 0.25, timeout: Optional[float] = None
+    ):
+        """Wait until Multipass is ready (upon install/startup).
+
+        :param retry_wait: Time to sleep between retries.
+        :param timeout: Optional time.time() stamp to timeout after.
+        """
+        while timeout is None or time.time() < timeout:
+            multipass_version, multipassd_version = self.version()
+
+            if multipassd_version is not None:
+                return (multipass_version, multipassd_version)
+
+            time.sleep(retry_wait)
+
+        raise MultipassError(
+            brief="Timed out waiting for Multipass to become ready.",
+        )
+
+    def version(self):
```

Comment:
return type

Reply:
Fixed.  Will need to revisit with another PR for stricter mypy config.  This should be caught automatically.